NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -510,21 +511,35 @@ class NeuroID
             }
 
             // immediately flush events before anything else
-            CoroutineScope(Dispatchers.IO).launch {
+            CoroutineScope(dispatcher).launch {
                 nidJobServiceManager.sendEvents(true)
                 saveIntegrationHealthEvents()
             }
+
+            linkedSiteID = siteID
```

Comment:
originally this was set early so startSession() / start() would process the linkedSiteID however now that we don't alter the state of the SDK, just stop event collection at capture() without stoping the SDK, this can go back to the original position. start(). startSession() and startAppFlow() don't need any changes.